### PR TITLE
Use tsup for building partysocket and y-partykit

### PR DESCRIPTION
**Pull request by mellson**
To make building and bundling less error-prone @Andarist and I looked at some options today. And we settled on [`tsup`](https://tsup.egoist.dev/) as a sane choice for now. We also looked briefly at [`preconstruct`](https://github.com/preconstruct/preconstruct), but that seemed to conflict a bit with the monorepo setup of this repo.

While this is not perfect, mainly because the `.d.ts` files need to be symlinked, it seems to work. So good enough for me :blush:

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,5 +2,11 @@
   "typescript.tsdk": "node_modules/typescript/lib",
   "editor.codeActionsOnSave": {
     "source.organizeImports": false
-  }
+  },
+  "json.schemas": [
```
[the remark this answers is not in the document]
Comment (Andarist) [reply .vscode/settings.json:6]: neat, TIL

**Comment by mellson** [comment]
> I'll do another small commit later today to remove the symlinking. @Andarist reminded me that the build previously had `mv dist/*.d.ts* .` - which I think is nicer. So I'll revert to that once I'm home

Done, ready for review.

**Review by Andarist — APPROVED** [review]
LGTM

**Review by threepointone — APPROVED** [review]
You beautiful geniuses

**Comment by mellson** [comment]
I can confirm that this version works as expected in our codebase, and it also looks like [the types are right](https://arethetypeswrong.github.io/?p=partysocket%400.0.0-cc3d569) 🎉

**Comment by Andarist** [comment]
to be completely fair... :P they are not completely right but i guess that one brings masques to a good party... so it checks out 😉

**Pull request merged**